atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -170,6 +172,24 @@ def validate_platforms_in_base_image(self, base_image):
                           'because platform descriptors are not defined')
             return
 
+        if '@sha256:' in str(base_image):
+            base_image_result = ImageName.parse(str(base_image))
+            base_image = base_image_result.copy()
```

Comment:
What's this for? I don't understand it. Isn't `base_image` an ImageName here? And why do we need to copy it?

Reply:
yes base_image is indeed ImageName, but look at line 191, I want to change tag to manifest_list nvr, but JUST for get_manifest_list, IF I'd change directly base_image without copying it, it would try later to pull manifest_list instead of that specified sha, which I want to keep unchanged for pulling

Reply:
So should these two lines be replaced by one?:
```
base_image = base_image.copy()  # we want to adjust the tag only for manifest list fetching
```

Reply:
right yes, will do that